lib/plugins/aws/package/compile/events/cloudFront.js

```diff
@@ -432,6 +432,13 @@ class AwsCompileCloudFrontEvents {
             };
             let shouldAssignCachePolicy = true;
             if (event.cloudFront.behavior) {
+              if (event.cloudFront.behavior.CachePolicyId) {
+                if (!event.cloudFront.cachePolicy) {
+                  event.cloudFront.cachePolicy = {
```

Comment:
Instead of mutating configuration of the event we should set that property directly on `behavior` object that will be passed to CloudFormation resource